src/NServiceBus.Core/Diagnostics/ContextPropagation.cs

```diff
@@ -0,0 +1,61 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+
+    static class ContextPropagation
```

Comment:
Should we consider pulling this out into an `IContextPropagationStrategy`? This would be the default implementation. This would make it easier for us to create a `NoOpContextPropagationStrategy` and have config methods to disable propagation. It could make it easier for us to allow a customer could provide their own implementation (although we may want to wait on that until we see evidence that it's needed).

Reply:
would definitely be an option but I'm not sure how much we would gain by that at the moment since this form of extensibility might not really be needed anyway?